internal/cli/build.go

```diff
@@ -223,6 +227,17 @@ func BuildCmd(ctx context.Context, imageRef, outputTarGZ string, archs []types.A
 				return fmt.Errorf("failed to build layer image: %w", err)
 			}
 
+			if _, ok := os.LookupEnv("SOURCE_DATE_EPOCH"); !ok {
```

Comment:
I think that this chunk of code should be in its own reusable func, as you reproduce it below in `cli/publish.go` as well.

Although I am wondering if this shouldn't just be an option we pass (likely in `pkg/build/options.go`):

```go
func WithInstallFilesTimestamp(ts string) ...
```

Where that says, "calculate the latest date and use it for anything that you add in". 

What we have here is duplicated in `cli/publish.go`, and then goes for the same variable in `pkg/build/build.go` [here](https://github.com/chainguard-dev/apko/blob/a89818861f525e38332c6caac33a79003598236d/pkg/build/build.go#L187) (which really is not good; we should be looking up vars here in `internal/cli` and not in `pkg/build`).

Reply:
I think we cleared up a bunch of confusion in slack, but this code could definitely be clearer and cleaner still.

I'll refactor it a bit, and add a bunch of comments because my [ab]use of `SourceDateEpoch` to tunnel the configuration timestamp through is subtle.